test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
After looking into it, there's not really a nice way to initialize a Termios object in a platform-agnostic manner until it implements `Default`, so I'd argue for pushing that fix later in a separate PR